assets/tests/util/result.test.ts

```diff
@@ -0,0 +1,48 @@
+import { describe, expect, jest, test } from "@jest/globals"
+import { Err, Ok, Result, isErr, isOk, map } from "../../src/util/result"
+
+describe("Result<T, E>::isOk", () => {
+  test("returns true for Ok<T>", () => {
+    expect(isOk(Ok(undefined))).toBe(true)
+  })
+  test("returns false for Err<E>", () => {
+    expect(isOk(Err(undefined))).toBe(false)
+  })
+
+  test("when `isOk`, allows accessing the internal value", () => {
+    const value = Ok(10) as Result<number, unknown>
+
+    expect(isOk(value) && value.ok).toEqual(10)
+  })
+})
+
+describe("Result<T, E>::isErr", () => {
+  test("returns true for Err<E>", () => {
+    expect(isErr(Err(undefined))).toBe(true)
+  })
+  test("returns false for Ok<T>", () => {
+    expect(isErr(Ok(undefined))).toBe(false)
+  })
+
+  test("when `isErr`, allows accessing the internal value", () => {
+    const value = Err(10) as Result<unknown, number>
+
+    expect(isErr(value) && value.err).toEqual(10)
+  })
+})
+
+describe("Result<T, E>::map", () => {
+  test("applies function if Ok<T>", () => {
+    const fn = jest.fn(() => true)
```

Comment:
Nit, non-blocking

Hmmm - any reason not to just test this with a real function? So something like

```typescript
const result: Result<number, unknown> = Ok(3)

expect(map(result, (x) => x * 2)).toEqual(Ok(6))
```

It's not like... some big objective truth or anything, just that it feels more natural to me to test this way (I'm not a huge fan of mocks, especially when non-mocked things work just fine).

Reply:
IMO, this is "real function" it just also comes with tracking abilities so we can assert what it's called with. It's just a personal preference to be able to assert that the function was called so I'm okay with leaving it off.

What do you think of [`d8230eb` (#2531)](https://github.com/mbta/skate/pull/2531/commits/d8230eba8735ed5ba1a4d551cc24192ce46776a8)?
I like it because it's completely impossible for `map` to know the value of `value` ahead of time, and this also tests that `map` allows for transforming the _type_ contained within `Ok<T>`

Reply:
Works for me! And I do like the idea of changing the type 👍

Reply:
I'd be tempted to try giving the other test a similar treatment, but as before - this is non-blocking.